subsys/net/lib/sockets/sockets.c

```diff
@@ -173,6 +175,76 @@ int z_impl_zsock_socket(int family, int type, int proto)
 	return -1;
 }
 
+int z_impl_zsock_socketpair(int family, int type, int proto, int sv[2]) {
+
+	int res;
+
+	/* Use a temporary socket vector since sv[] should not be modified
+         * on failure. See POSIX.1-2016 */
+	int temp_sv[2];
+
+	/* Zero-initializing this (with the exception of setting the family)
+	 * has the same effect as using INADDR_ANY_INIT or IN6ADDR_ANY_INIT.
+         *
+	 * The address family field is set below, in the switch statement.
+	 */
+	union {
+		struct sockaddr_in  in;
+		struct sockaddr_in6 in6;
+	} addr = {0};
+
+	struct sockaddr *sa;
+	socklen_t len;
+
+	switch(family) {
+	case AF_INET:
+		sa = (struct sockaddr *) &addr.in;
+		sa->sa_family = AF_INET;
+		len = sizeof(addr.in);
+		break;
+	case AF_INET6:
+		sa = (struct sockaddr *) &addr.in6;
+		sa->sa_family = AF_INET6;
+		len = sizeof(addr.in6);
+		break;
+	default:
+		errno = EAFNOSUPPORT;
+		return -1;
+	}
+
+	res = z_impl_zsock_socket(family, type, proto);
+	if (-1 == res) {
+		return -1;
+	}
+	temp_sv[0] = res;
+
+	res = z_impl_zsock_socket(family, type, proto);
+	if (-1 == res) {
```

Comment:
Please write this `if (res == -1) {` in order to follow the same style as the other functions in this file.